lib/persist.js

```diff
@@ -1,50 +1,66 @@
 require("dotenv").config();
-const redis = require("redis").createClient(process.env.REDIS_URL);
-const { promisify } = require("util");
-const get = promisify(redis.get).bind(redis); // this maps redis.get(key, callback(results)) to a promise-compatible function
+
 const moment = require("moment");
 const { getCurrentEmojis } = require("./slack");
 
+const makeRedisClient = async () => {
+  const redis = require("redis").createClient({
+    url: process.env.REDIS_URL,
+  });
+  redis.on("error", (err) => console.log("Redis Client Error", err));
+  await redis.connect();
+
+  return redis;
+};
+
 const FORBIDDEN_DAYS_OF_WEEK = ["Saturday", "Sunday"];
 
-const getLastKnownEmoji = async (todayIso) => {
-  const week = await Promise.all(
-    Array.from(Array(7)).map((_, idx) => {
-      const key = moment(todayIso)
-        .subtract(idx + 1, "day")
-        .format("MM-DD-YYYY");
-      return get(key);
-    })
-  );
+const getLastKnownEmoji = async () => {
+  const redis = await makeRedisClient();
+  const last_known = await redis.get("last_known");
 
-  const lastKnown = week.find((day) => !!day);
+  if (!last_known) {
+    return null;
+  }
 
-  return JSON.parse(lastKnown);
+  return JSON.parse(last_known);
 };
 
-const synchronizeEmojis = async () => {
```

Comment:
this was also cursed code. `synchronizeEmojis` used to look up the last time we ran emoji bot, get the entire list of emoji at that time, and then returning both to the digest to be diff'd. this was also for some reason responsible for the business rule of running/not running on the weekend?